ui/app/js/controllers/user_edit_controller.js

```diff
@@ -5,6 +5,88 @@ function ($scope, $modalInstance, CSRF, Permissions, users, is_edit, user) {
   $scope.loading = true;
   $scope.users = users;
   $scope.currentItemTab = 1;
+
+  var allPermissions = {
+        "admin": ["products"],
+        "release": ["actions", "products"],
+        "release_locale": ["actions", "products"],
+        "release_read_only": ["actions", "products"],
+        "rule": ["actions", "products"],
+        "permission": ["actions"],
+        "required_signoff": ["products"],
+        "scheduled_change": ["actions"],
+    };
+
+  $scope.onPermission = function(){
+      $scope.available_options = allPermissions[$scope.permission.permission];
+
+      // clear  all previous selections 
+      [
+          $scope.selected_options,
+          $scope.selected_actions,
+          $scope.selected_products
+      ] = [[], [], []];
```

Comment:
This would be easier to read if it was written more plainly as:
$scope.selected_options = [];
etc.

Reply:
@mozbhearsum Thanks a lot for the resource, you just saved my day. Had spent lots hours trying to figure out all the possible options. 👍